system/jlib/jqueue.tpp

```diff
@@ -33,7 +33,7 @@ class QueueOf
     unsigned headp;
     unsigned tailp;
     unsigned max;
-    unsigned num;
+    RelaxedAtomic<unsigned> num; // atomic so that it can be read without a critical section
```

Comment:
Seems to me that ordinality() is always going to give an answer that is only guaranteed valid for the duration of the critical block in which it was read. Removing the CriticalBlock is a good change, but making the variable RelaxedAtomic seems therefore pointless.

Reply:
The only reason it is a relaxed atomic is to guarantee that the value is read (to conform correctly to the c++ standard)